### DISCUSSION TITLE
How do I extract all the elements on the grain boundaries?

### DISCUSSION THREAD

**Discussion by smallrain1996** [neperfepx/neper#750] [Q&A] [answered]
Is there a way to output the ids of all elements located on the grain boundaries? If you can tell me what command to use to create an Elset composed of all elements in the model that are located at the grain boundary, or how to output a statistical file that shows whether each element is an in-grain element or a grain boundary element?

**Comment by mkasemer** [comment, marked as answer]
Try: `neper -M … -statelt elsetbody`.

To be clear: this will contain `0` for elements which are on the grain boundary, `1` for elements which are separated from the grain boundary by a single layer of elements, etc.

To a question below, use `neper -M ... -statelt body` to find the same information relative to the domain, rather than the grain boundaries.

**Reply by smallrain1996** [reply]
Thank you very much for your reply.
I tried, but it didn't work.
'test_neper_3D.stelt' all 0.
```
name=test_neper_3D
number=150
mkdir $name; cd $name
neper -T -n $number \
-format tess -o $name
neper -M $name.tess -format inp,msh,phase \
-nset faces -faset none \
-statelt elsetbody \
-elttype hex -rcl 0.8 \
-o $name
neper -V $name.msh -datacellcol id -print $name
```

**Reply by mkasemer** [reply]
What version of Neper? Have you updated recently?

It is also possible that your mesh is not of sufficient resolution to find any elements that do not touch the grain boundary. I would suspect that `-rcl 0.8` should give you a mesh in which there are true grain-interior elements, but unknown. 

Alternatively, but in the same vein, is your `stelt` file truly a vector of all 0s, or is just that there are so few that you are not noticing them? Load into Python or MATLAB and check the number of non-zero values.

**Reply by smallrain1996** [reply]
I'm using the latest version of Neper, so you can try executing my code. It's true that all the elements are 0, and I checked with the tool.

**Reply by mkasemer** [reply]
Try refining the mesh further. Otherwise, @rquey will have to comment.

**Reply by smallrain1996** [reply]
Maybe it was a Bug, I used a mesh like 100 100 100, and so on.

**Reply by rquey** [reply]
It won't work with hex meshing...

**Reply by smallrain1996** [reply]
Crying instantly, this is really a very sad thing.

**Comment by charanjo** [comment]
Is there a way to extract elements on the surface of the domain?

**Reply by mkasemer** [reply]
Try: `neper -M … -statelt body`
